### PR TITLE
[REFACT] - lambda service invoke function

### PR DESCRIPTION
**Pull request by marcosalves-bitg**
This PR just refacts the Lambda Invoke function, to use just one.

We divided the invokeLambda function into two, to use one with a custom secret and another (default) passing no secret value.
I refact the two functions into one

### CODE REVIEW

**Comment by marcosalves-bitg** [comment]
Guys! I requested your reviews so we can discuss this again. We, at BombSquad, were not happy with the two functions approach.

**Comment by marcosalves-bitg** [comment]
PS: We need to improve the version. Is this a Major or a Minor? Or just a Patch?

**Comment by th3rius** [comment]
> PS: We need to improve the version. Is this a Major or a Minor? Or just a Patch?

Major; this is a breaking change

**Review by Viserion77 — DISMISSED** [review]
LGTM

**Comment by marcosalves-bitg** [comment]
> > PS: We need to improve the version. Is this a Major or a Minor? Or just a Patch?
> 
> Major; this is a breaking change

I agree

File: src/lambda/interfaces.ts [hunk 1]
```diff
@@ -32,3 +33,8 @@ export type CrudInputParams = {
   keys?: {[key: string]: any}
   customParameters?: {[key: string]: any}
 }
+
+export interface SecretValue {
```
Comment (mickaelnn) [inline src/lambda/interfaces.ts:37]: The Secret Value can be an object of any type, the "SecretValue" maybe can be named as "AccessKey" and contain region in this secret also, but as optional

File: src/lambda/lambdaService.ts [hunk 2]
```diff
@@ -2,38 +2,29 @@ import AWS from 'aws-sdk'
...
 export class LambdaService {
   static async invoke (lambdaParameters: lambdaParameters): Promise<object> {
     try {
-      return executeInvoke(lambdaParameters)
-    } catch (error) {
-      console.log('Error LambdaService invoke', error)
-      throw error
-    }
-  }
+      const secretValue: SecretValue = lambdaParameters.serviceSecretArn
```
Comment (mickaelnn) [inline src/lambda/lambdaService.ts:11]: here also can be named as accessKey...

**Review by mickaelnn — CHANGES_REQUESTED** [review]

File: src/lambda/interfaces.ts [hunk 3]
```diff
@@ -32,3 +33,9 @@ export type CrudInputParams = {
   keys?: {[key: string]: any}
   customParameters?: {[key: string]: any}
 }
+
+export interface AccessKey {
+  accessKeyId?: string
+  secretAccessKey?: string
```
Comment (mickaelnn) [inline src/lambda/interfaces.ts:39]: this is a mandatory prop

**Review by mickaelnn — CHANGES_REQUESTED** [review]

**Comment by mickaelnn** [comment]
I think this function is an invoke function, this code to get accessKey from secrets manager is another function, and if needed to use with another accessKey, you need to use this by params directly to invoke

obs: this is not the normal flow from an invoke function...

File: src/lambda/lambdaService.ts [hunk 4]
```diff
@@ -2,38 +2,29 @@ import AWS from 'aws-sdk'
...
-      return executeInvoke(lambdaParameters)
-    } catch (error) {
-      console.log('Error LambdaService invoke', error)
-      throw error
-    }
-  }
+      const accessKey: AccessKey = lambdaParameters.serviceSecretArn
+        ? await SecretManager.getValue({
+          region: lambdaParameters.region,
+          secretId: lambdaParameters.serviceSecretArn
```
Comment (alanneves) [inline src/lambda/lambdaService.ts:14]: na minha visão, a lógica do secretmaneger não deve ficar no invoke
o invoke pode receber accessKey, não secretARN, o get do secretARN deveria ser feito antes e só se passado o acesskey

o invoke com  essa lógica do secretManger quebra [Single-responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle)

**Review by alanneves — CHANGES_REQUESTED** [review]

**Review by mickaelnn — APPROVED** [review]
LGTM

**Review by alanneves — DISMISSED** [review]
removendo request changes solicitado pelo rica

**Review by ricardobaehr — APPROVED** [review]
LGTM

**Pull request merged**